src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,125 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int factCapacity;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        factCapacity = INITIAL_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
```

Comment:
There are a few areas where your code can be improved. Let's go through them:

Code duplication: There's a repetition of the same if condition `if (node.hash == hash && Objects.equals(node.key, key))` thrice in your method. This code duplication makes your method harder to maintain and understand. This can be solved by restructuring the logic of your while loop. If the keys are equal, then just replace the value and return, and only if you've traversed through the entire bucket (and not found a matching key), you should add the new node.

Unnecessary hash comparison: You're comparing hashes of the keys, which is unnecessary. The bucket index is already calculated based on the hash of the key. Once you are within a bucket, you can simply compare the keys themselves.

To make your put method more succinct, you could follow this algorithm:

Calculate the index from the key.
If the bucket at that index is empty, create a new node, place it in the bucket, and increment the size.
If the bucket is not empty, iterate through the nodes in the bucket.
If you find a node with a key that equals the key you're trying to put, replace its value.
If no such node is found (you've reached the end of the nodes in the bucket), add a new node at the end and increment the size.